docs/l3/api_flow.md

```diff
@@ -0,0 +1,413 @@
+      "max_width": 100
+    },
+    "item_dimensions": {
+      "length": 80,
+      "height": 50,
+      "width": 40
+    },
+    "item_weight": 25,
+    "pickup_location": {
+      "geo_location": {
+        "latitude": 51.5074,
+        "longitude": -0.1278
+      },
+      "timed_window": {
+        "start_time": "2023-05-10T10:00:00Z",
+        "end_time": "2023-05-10T12:00:00Z"
+      },
+      "expected_time_for_loading": 30
+    },
+    "dropoff_location": {
+      "geo_location": {
+        "latitude": 52.5200,
+        "longitude": 13.4050
+      },
+      "estimated_time": "2023-05-11T15:30:00Z"
+    },
+    "mixing_exclusion_criteria": {
+      "critical": [
+        "toxic",
+        "flammable"
+      ],
+      "user_defined": [
+        "organic",
+        "inorganic"
+      ]
+    },
+    "specialist_requirements": [
+      "refrigerated",
+      "hazardous"
+    ]
+  }
+```
+#### Member B
+```
+
+```
+
+Member A uploads the parameters file for `demand_a` to their local database using `POST /v1/attachment`.
+
+Response:
+```
+{
+  "id": "78b6a81c-5ef3-4549-82a9-e9459ca0cc93",
+  "filename": "attachment_order.json",
+  "size": 999,
+  "createdAt": "2023-05-09T14:54:14.708Z"
+}
+```
+
+Member A creates `demand_a` by sending a `POST request to /v1/demandA`, including the `id` for the parameters file in the request body. 
+
+Response:
+```
+{
+  "id": "18f832d2-162f-486c-bbc6-d4c436df0d63",
+  "owner": "Member A",
+  "state": "created",
+  "parametersAttachmentId": "78b6a81c-5ef3-4549-82a9-e9459ca0cc93",
+  "createdAt": "2023-05-09T14:58:17.408Z",
+  "updatedAt": "2023-05-09T14:58:17.408Z"
+}
+```
+
+When Member A is ready for `demand_a` to exist on the blockchain, they send a `POST request to /v1/demandA/{demandAId}/creation`. 
+
+Response:
+```
+{
+  "id": "8c783421-eb61-4bdb-b719-aee945d02121",
+  "state": "submitted",
+  "localId": "18f832d2-162f-486c-bbc6-d4c436df0d63",
+  "apiType": "demand_a",
+  "transactionType": "creation",
+  "submittedAt": "2023-05-09T15:04:19.422Z",
+  "updatedAt": "2023-05-09T15:04:19.422Z"
+}
+```
+
+Member A receives a transaction id for the `demand_a` creation transaction, which they can use to get the details of the transaction using `GET /v1/demandA/{demandAId}/creation/{creationId}`. They can also use `GET /v1/transaction` to get all transactions of any type. 
+
+#### GET /v1/demandA/{demandAId}/creation/{creationId}
+
+Inputting the `demand_a`'s identifier as: `18f832d2-162f-486c-bbc6-d4c436df0d63`
+
+Inputting the `demand_a`'s creation ID as: `8c783421-eb61-4bdb-b719-aee945d02121`
+
+Response:
+```
+{
+  "id": "8c783421-eb61-4bdb-b719-aee945d02121",
+  "state": "finalised",
+  "localId": "18f832d2-162f-486c-bbc6-d4c436df0d63",
+  "apiType": "demand_a",
+  "transactionType": "creation",
+  "submittedAt": "2023-05-09T15:04:19.422Z",
+  "updatedAt": "2023-05-09T15:04:37.227Z"
+}
+```
+
+#### GET /v1/transaction
+
+Inputting the apiType as: `demand_a`
+
+Inputting the status as: `finalised`
```

Comment:
could make it clearer that these are URL params like `/v1/transaction?apiType=demand_a&status=finalised`

Reply:
```suggestion
Inputting the apiType as: `demand_a`

Inputting the status as: `finalised`

URL parameters: `/v1/transaction?apiType=demand_a&status=finalised`
```